src/components/OfflineSearch.ts

```diff
@@ -0,0 +1,174 @@
+import { Component, createElement } from "react";
+import { findDOMNode } from "react-dom";
+
+import * as dijitRegistry from "dijit/registry";
+
+import { Alert } from "./Alert";
+import "../ui/OfflineSearch.css";
+
+type SearchMethodOptions = "equals" | "contains";
+
+interface ListView extends mxui.widget._WidgetBase {
+    _datasource: {
+        _constraints: string | null;
+        _setSize: number;
+        atEnd: () => boolean;
+        _pageSize: number;
+    };
+    _loadMore: () => void;
+    _onLoad: () => void;
+    _renderData: () => void;
+    update: () => void;
+}
+
+interface OfflineSearchProps {
+    defaultQuery: string;
+    searchEntity: string;
+    searchAttribute: string;
+    showSearchBar: boolean;
+    targetGridName: string;
+    searchMethod: SearchMethodOptions;
+}
+
+interface OfflineSearchState {
+    alertMessage?: string;
+    buttonVisibility?: string;
+}
+
+export default class OfflineSearch extends Component<OfflineSearchProps, OfflineSearchState> {
+    // internal variables
+    private searchButton: HTMLButtonElement;
+    private searchInput: HTMLInputElement;
+    private targetWidget: ListView;
+
+    constructor(props: OfflineSearchProps) {
+        super(props);
+
+        this.state = {
+            alertMessage: "",
+            buttonVisibility: "hidden"
+        };
+        this.onClear = this.onClear.bind(this);
+        this.onSearchKeyDown = this.onSearchKeyDown.bind(this);
+    }
+
+    render() {
+        return (
+        createElement("div", { className: "widget-offline-search" },
+            createElement(Alert, { className: "danger", message: this.state.alertMessage }),
+            this.props.showSearchBar
+            ?
+            createElement("div", { className: "search-container" },
+                createElement("span", { className: "glyphicon glyphicon-search" }),
+                createElement("input", {
+                    className: "form-control", placeholder: "Search",
+                    ref: input => this.searchInput = input as HTMLInputElement
+                }),
+                createElement("button", {
+                    className: `btn-transparent ${this.state.buttonVisibility}`,
+                    ref: button => this.searchButton = button as HTMLButtonElement
+                },
+                    createElement("span", { className: "glyphicon glyphicon-remove" })
+                )
+            )
+            : createElement("div", { className: "search-container-hidden" })
```

Comment:
Only significance i see is when we start doing end to end tests/selenium. It will come in handy when a test if being written to check if the widget does not show if status but still it can be null